tfjs-backend-webgpu/src/kernels/binary_op_util.ts

```diff
@@ -116,42 +102,180 @@ const MUL = 'return a * b;';
 const NOT_EQUAL = 'return float(a != b);';
 const NOT_EQUAL_VEC4 = 'return vec4(notEqual(a, b));';
 const POW = `
-if(a < 0.0 && floor(b) < b){
-  return NAN;
-}
-if (b == 0.0) {
-  return 1.0;
-}
-return (round(mod(b, 2.0)) != 1) ?
-    pow(abs(a), b) : sign(a) * pow(abs(a), b);
-`;
+  if(a < 0.0 && floor(b) < b) {
+    return NAN;
+  }
+  if (b == 0.0) {
+    return 1.0;
+  }
+  return (round(mod(b, 2.0)) != 1) ?
+      pow(abs(a), b) : sign(a) * pow(abs(a), b);
+  `;
 const POW_VEC4 = `
-// isModRound1 has 1 for components with round(mod(b, 2.0)) == 1, 0 otherwise.
-vec4 isModRound1 = vec4(equal(round(mod(b, 2.0)), ivec4(1)));
-vec4 multiplier = sign(a) * isModRound1 + (vec4(1.0) - isModRound1);
-vec4 result = multiplier * pow(abs(a), b);
+  // isModRound1 has 1 for components with round(mod(b, 2.0)) == 1, 0 otherwise.
+  vec4 isModRound1 = vec4(equal(round(mod(b, 2.0)), ivec4(1)));
+  vec4 multiplier = sign(a) * isModRound1 + (vec4(1.0) - isModRound1);
+  vec4 result = multiplier * pow(abs(a), b);
 
-// Ensure that a^0 = 1, including 0^0 = 1 as this correspond to TF and JS
-bvec4 isExpZero = equal(b, vec4(0.0));
-result.r = isExpZero.r ? 1.0 : result.r;
-result.g = isExpZero.g ? 1.0 : result.g;
-result.b = isExpZero.b ? 1.0 : result.b;
-result.a = isExpZero.a ? 1.0 : result.a;
+  // Ensure that a^0 = 1, including 0^0 = 1 as this correspond to TF and JS
+  bvec4 isExpZero = equal(b, vec4(0.0));
+  result.r = isExpZero.r ? 1.0 : result.r;
+  result.g = isExpZero.g ? 1.0 : result.g;
+  result.b = isExpZero.b ? 1.0 : result.b;
+  result.a = isExpZero.a ? 1.0 : result.a;
 
-vec4 isNaN = vec4(lessThan(a, vec4(0.0))) * vec4(lessThan(floor(b), b));
-${CHECK_NAN_SNIPPET_VEC4}
-return result;
-`;
+  vec4 isNaN = vec4(lessThan(a, vec4(0.0))) * vec4(lessThan(floor(b), b));
+  ${CHECK_NAN_SNIPPET_VEC4}
+  return result;
+  `;
 const PRELU = 'return (a < 0.) ? b * a : a;';
 const PRELU_VEC4 = `
-vec4 aLessThanZero = vec4(lessThan(a, vec4(0.)));
-return (aLessThanZero * (b * a)) + ((vec4(1.0) - aLessThanZero) * a);
-`;
+  vec4 aLessThanZero = vec4(lessThan(a, vec4(0.)));
+  return (aLessThanZero * (b * a)) + ((vec4(1.0) - aLessThanZero) * a);
+  `;
 const SQUARED_DIFFERENCE = 'return (a - b) * (a - b);';
 const SUB = 'return a - b;';
 
+// WGSL shader.
+const EQUAL_WGSL = 'if (a == b) { return 1.0; } return 0.0;';
```

Comment:
If https://github.com/gpuweb/gpuweb/issues/1910 is supported by chrome canary, we can use the same code snippet for `EQUAL` between glsl and wgsl? Please also check other places.

Reply:
Related workaround removed, PTAL.